src/FindNetworkEndpoint/index.ts

```diff
@@ -0,0 +1,50 @@
+import { INetwork } from './interface';
+
+export class FindNetworkEndpoint {
+  network: INetwork;
+  startNodeId: number;
+
+  constructor(startNodeId: number, fromIds: number[], toIds: number[]) {
+    if (fromIds.length !== toIds.length)
+      throw Error(`${fromIds} and ${toIds} must have the same length`);
+    if (fromIds.length < 1 || toIds.length < 1)
+      throw Error(`${fromIds} and ${toIds} must not be empty arrays`);
+    this.network = this.convertToPath(fromIds, toIds);
+    this.startNodeId = startNodeId;
+  }
+
+  private convertToPath(fromIds: number[], toIds: number[]): INetwork {
+    const network: INetwork = {};
+    fromIds.map((fromId, index) => {
+      network[fromId] = toIds[index];
+    });
+    return network;
+  }
+
+  find(): number {
+    let endNode = this.startNodeId;
+    if (this.network[this.startNodeId] === undefined) {
+      throw Error(`${this.startNodeId} is not existing in ${this.network}`);
+    } else {
+      const networkArrayObject = Object.entries(this.network).splice(
+        Object.keys(this.network).indexOf(this.startNodeId.toString())
+      );
+
+      for (const [nodeId, nodeValue] of networkArrayObject) {
+        if (parseInt(nodeId) === endNode) {
+          const lastNodeValue = Object.values(this.network).pop();
+          const firstNodeId = Object.keys(this.network).shift();
```

Comment:
I do not understand this part what is happening here?

Reply:
to avoid infinite loop.